samples/openvx/wrappers.cpp

```diff
@@ -0,0 +1,214 @@
+        Image::Patch resultPatch;
+        Mat output;
+        if (useCopy || mapFromVx)
+        {
+            //we will copy or map data from vx_image to cv::Mat
+            ivxResult = ivx::Image::create(context, width, height, VX_DF_IMAGE_U8);
+        }
+        else if (!mapFromVx)
+        {
+            //create vx_image based on user data, no copying required
+            output = cv::Mat(height, width, CV_8U, cv::Scalar(0));
+            vx_imagepatch_addressing_t addressing = Image::createAddressing(width, height, (vx_uint32)output.elemSize(),
+                                                                            (vx_uint32)output.step);
+            const std::vector<vx_imagepatch_addressing_t> addrs(1, addressing);
+            const std::vector<void*> ptrs(1, output.data);
+            ivxResult = Image::createFromHandle(context, Image::matTypeToFormat(CV_8U), addrs, ptrs);
+        }
+
+        Graph graph = createProcessingGraph(context, ivxImage, ivxResult);
+
+        // Graph execution
+        graph.process();
+
+        //getting resulting image in cv::Mat
+        if (useCopy)
+        {
+            ivxResult.copyTo(0, output);
+        }
+        else
+        {
+            if (mapFromVx)
+            {
+                //create cv::Mat based on vx_image mapped data
+                resultPatch.map(ivxResult, 0, ivxResult.getValidRegion(), VX_READ_AND_WRITE);
+                output = resultPatch.getMat();
+            }
+            else
+            {
+#ifdef VX_VERSION_1_1
+                //we should take user memory back from vx_image before using it (even before reading)
+                ivxResult.swapHandle();
+#endif
+            }
+        }
+
+        //here output goes
+        cv::imshow("processing result", output);
+        cv::waitKey(0);
+
+        cv::destroyAllWindows();
+
+#ifdef VX_VERSION_1_1
+        if (!useCopy)
+        {
+            //we should take user memory back before release
+            //(it's not done automatically according to standard)
+            ivxImage.swapHandle();
+            if (!mapFromVx) ivxResult.swapHandle();
+        }
+#endif
+
+        //the string is unnecessary since unmapping is done on destruction of patch
+        //resultPatch.unmap();
+    }
+    catch (const ivx::RuntimeError& e)
+    {
+        std::cerr << "Error: code = " << e.status() << ", message = " << e.what() << std::endl;
+        return e.status();
+    }
+    catch (const ivx::WrapperError& e)
+    {
+        std::cerr << "Error: message = " << e.what() << std::endl;
+        return -1;
+    }
+
+    return 0;
+}
+
+
+int main(int argc, char *argv[])
+{
+    const std::string keys =
+        "{help h usage ? | | }"
+        "{@image_name    | | image to be processed}"
+        "{nocopy         | | use handles to user-allocated memory instead of creating input/output VX images and copying data to/from them }"
+        "{map            | | map resulting VX image to user memory }"
```

Comment:
I suggest joining both 'nocopy' and 'map' to a single option named 'mode' with values like 'copy', 'map' and 'user_mem' or 'handle'